app/models/article.rb

```diff
@@ -1,4 +1,25 @@
 class Article < ApplicationRecord
     validates :title, presence: true, uniqueness: true
     validates :body, presence: true
+    validates :slug, uniqueness: true
+    
+    after_create :update_slug
+    before_update :assign_slug

```

Comment:
Seems weird that before_**update** is assign_slug and before_**create** is **update**_slug.

Reply:
Also, you are using the private method `assign_slug` but that method calls public method `create_slug`. Do you need `create_slug` to be public? If not, you could probably put it’s code directly in-line inside of `assign_slug`.

Reply:
Lastly, the purpose of a slug is to have a human readable but never changing url, right? Your existing code updates the slug every time you change the title. Links with old slugs won’t work anymore.